File: Day126_2025/Q126_GFG.py
class Solution:
    def LeftView(self, root):
        result = []

        def leftViewUtil(node, level):
            if not node:
                return
            if len(result) == level:
                result.append(node.data)
            leftViewUtil(node.left, level + 1)
            leftViewUtil(node.right, level + 1)

        leftViewUtil(root, 0)
        return result

File: Day126_2025/Q126_GFG.java
import java.util.ArrayList;

class Q126_GFG {
    void leftViewUtil(Node root, int level, ArrayList<Integer> result) {
        if (root == null) return;
        
        if (result.size() == level) {
            result.add(root.data);
        }
        
        leftViewUtil(root.left, level + 1, result);
        leftViewUtil(root.right, level + 1, result);
    }

    ArrayList<Integer> leftView(Node root) {
        ArrayList<Integer> result = new ArrayList<>();
        leftViewUtil(root, 0, result);
        return result;
    }
}
